runelite-client/src/test/java/net/runelite/client/util/ImageUtilTest.java

```diff
@@ -0,0 +1,437 @@
+	}
+
+	@Test
+	public void alphaOffset()
+	{
+		// alphaOffset(BufferedImage image, int offset)
+		assert(bufferedImagesEqual(oneByOne(BLACK_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK_TRANSPARENT), -255)));
+		assertEquals(new Color(0, 0, 0, 50).getRGB(), ImageUtil.alphaOffset(oneByOne(BLACK_TRANSPARENT), 50).getRGB(0, 0));
+		assert(bufferedImagesEqual(oneByOne(BLACK_HALF_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK_TRANSPARENT), 128)));
+		assert(bufferedImagesEqual(oneByOne(BLACK_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK_HALF_TRANSPARENT), -255)));
+		assert(bufferedImagesEqual(oneByOne(BLACK), ImageUtil.alphaOffset(oneByOne(BLACK_TRANSPARENT), 255)));
+		assertEquals(new Color(0, 0, 0, 200).getRGB(), ImageUtil.alphaOffset(oneByOne(BLACK), -55).getRGB(0, 0));
+		assert(bufferedImagesEqual(oneByOne(BLACK), ImageUtil.alphaOffset(oneByOne(BLACK), 255)));
+
+		// alphaOffset(BufferedImage image, float offset)
+		assert(bufferedImagesEqual(oneByOne(BLACK_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK_TRANSPARENT), 0f)));
+		assert(bufferedImagesEqual(oneByOne(BLACK_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK_TRANSPARENT), 1f)));
+		assert(bufferedImagesEqual(oneByOne(BLACK_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK_TRANSPARENT), 2f)));
+		assert(bufferedImagesEqual(oneByOne(BLACK_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK_HALF_TRANSPARENT), 0f)));
+		assert(bufferedImagesEqual(oneByOne(BLACK_HALF_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK_HALF_TRANSPARENT), 1f)));
+		assert(bufferedImagesEqual(oneByOne(BLACK), ImageUtil.alphaOffset(oneByOne(BLACK_HALF_TRANSPARENT), 2f)));
+		assert(bufferedImagesEqual(oneByOne(BLACK_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK), 0f)));
+		assert(bufferedImagesEqual(oneByOne(BLACK_HALF_TRANSPARENT), ImageUtil.alphaOffset(oneByOne(BLACK), 0.503f))); // opacityOffset does Math.floor
+		assert(bufferedImagesEqual(oneByOne(BLACK), ImageUtil.alphaOffset(oneByOne(BLACK), 1f)));
+		assert(bufferedImagesEqual(oneByOne(BLACK), ImageUtil.alphaOffset(oneByOne(BLACK), 2f)));
+	}
+
+	@Test
+	public void grayscaleImage()
+	{
+		final BufferedImage[] grayscaleColors = new BufferedImage[] {
+			oneByOne(WHITE),
+			oneByOne(GRAY),
+			oneByOne(BLACK),
+			oneByOne(BLACK_HALF_TRANSPARENT),
+			oneByOne(BLACK_TRANSPARENT),
+		};
+		final BufferedImage[] nonGrayscaleColors = new BufferedImage[] {
+			oneByOne(RED),
+			oneByOne(GREEN),
+			oneByOne(BLUE),
+		};
+
+		for (BufferedImage image : grayscaleColors)
+		{
+			assert(isGrayscale(image));
+		}
+		for (BufferedImage image : nonGrayscaleColors)
+		{
+			assert(!isGrayscale(image));
+		}
+		for (BufferedImage image : ArrayUtils.addAll(grayscaleColors, nonGrayscaleColors))
+		{
+			assert(isGrayscale(ImageUtil.grayscaleImage(image)));
+		}
+	}
+
+	@Test
+	public void resizeImage()
+	{
+		// TODO: test image contents after changing size
```

Comment:
The scaling algorithm does not exactly match that of other programs (like Photoshop) so I wasn't able to effectively test this. If needed, I could use the method to create some resized versions of a skill icon (like the overall skill icon) and add tests to compare them, but that seemed rather useless to me.